torch/csrc/distributed/c10d/NCCLUtils.hpp

```diff
@@ -179,12 +179,20 @@ std::string getNcclErrorDetailStr(
 // Write NCCL debug info to local disk or any storage users define.
 class TORCH_API DebugInfoWriter {
  public:
-  DebugInfoWriter(int rank);
   virtual ~DebugInfoWriter();
   virtual void write(const std::string& ncclTrace);
+  static DebugInfoWriter& getWriter(int rank);
```

Comment:
making these methods static is OK.  but it is not required, and could spoil using the `logPrefix()` function for example.  It does not hurt to leave these as instance methods, which can still access the static variables but also can access any instance variables.

Reply:
Actually no... This is actually expected because we don't allow calling constructor directly for a singleton. that's why both get and set are static.